daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PrimitivesTextNumber1.scala

```diff
@@ -582,17 +581,18 @@ class NumberFormatFactoryStatic[S](context: ThrowsSDE,
       getGroupingSep(gse, context)
     }
 
-  val expRep = {
-    Assert.invariant(exponentRepExpEv.isConstant)
-    getExponentRep(exponentRepExpEv.maybeConstant.get, context)
-  }
+  val expRep =
+    if (exponentRepExpEv.isEmpty) Nope else One {
+      Assert.invariant(exponentRepExpEv.get.isConstant)
+      getExponentRep(exponentRepExpEv.get.maybeConstant.get, context)
+    }
```

Comment:
This is a really weird way to write this (I think I wrote the stuff this is based on). I guess it's best to be consistent with the already existing code that does something similar, but we should avoid this in the future.